### PR TITLE
Add Polkadot RPCs to the Dapp examples and wallet

### PR DESCRIPTION
**Pull request by hamidra**
I could not find any CONTRIBUTE docs so hope this is correct.

We are exploring adding wallet-connect support for Polkadot dapps/wallets.
The added Polkadot RPCs in these examples can work as an example implementation for having Wallet-connect support in Polkadot dapps wallets.

These changes:

- Adds Polkadot and its testnet(Westend) as well as RPCs tx and signing to the supported networks in stand-alone dapp example (react-dapp-v2).

- Adds polkadot accounts + message/tx  signing to the wallet

https://youtu.be/faAKiQ6555s

### CODE REVIEW

File: .gitignore
Comment (bkrem) [inline .gitignore:1]: Please remove, each example defines its own `.gitignore` so this shouldn't be needed at the root.

File: dapps/react-dapp-v2/src/chains/polkadot.ts
Comment (bkrem) [inline dapps/react-dapp-v2/src/chains/polkadot.ts:62]: Looks like something happened with string formatting in this file specifically? The other diffs still use double quote `""` so not sure what happened 😄

File: dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx
Comment (bkrem) [inline dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:116]: Leftover log

File: dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx
Comment (bkrem) [inline dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:629]: Nice 👍 thank you for context here

File: dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx
Comment (bkrem) [inline dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:644]: Is this referring to react-scripts in the dapp example? 

This example was refactored to next.js a few days ago so might be worth rebasing/merging from `main` if you were implementing against the old CRA setup 👍

File: dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx
Comment (bkrem) [inline dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:663]: Is this leftover or referring to something specific in this method?

File: wallets/react-wallet-v2/src/pages/index.tsx
Comment (bkrem) [inline wallets/react-wallet-v2/src/pages/index.tsx:34]:
```suggestion
        <AccountCard key={name} name={name} logo={logo} rgb={rgb} address={polkadotAddress} />
```

File: wallets/react-wallet-v2/src/pages/index.tsx
Comment (bkrem) [inline wallets/react-wallet-v2/src/pages/index.tsx:49]:
```suggestion
            <AccountCard key={name} name={name} logo={logo} rgb={rgb} address={polkadotAddress} />
```

File: dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx
Comment (bkrem) [reply dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:116]:
```suggestion

```

**Review by bkrem — CHANGES_REQUESTED** [review]
Great work here @hamidra, thank you for trailblazing Polkadot support in the example dapp and wallet 🙏 

Thank you also for the youtube video showcasing the new functionality 💯 

Some minor fixes required but I think we're good to go once they're resolved

**Review by IljaDaderko — COMMENTED** [review]
Looks good to me from wallet example perspective 👍 Thank you for keeping same overall structure as well!

File: dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx
Comment (hamidra) [reply dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx]: Right:) I initially made changes on CRA version but after merging  the Next version,  forgot it is not using CRA anymore! It works now and I added back the signature verification.

File: dapps/react-dapp-v2/src/chains/polkadot.ts
Comment (hamidra) [reply dapps/react-dapp-v2/src/chains/polkadot.ts]: It was my prettier which was set to single quote:) fixed.

File: dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx
Comment (hamidra) [reply dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx]: The column indeed is misplaced, but it is related to the "await cryptoWaitReady()". In order to use sr25519 signatures in the lib it needs to wait till relevant WASM code loads. I moved the comment to the correct line.

**Comment by hamidra** [comment]
Excellent! Thanks. I addressed the comments. 
I'm gonna talk to some polkadot wallet teams to implement the wallet-connect on their side. If any changes happen in the RPCs I will send PRs to keep these up-to-date, but should be good for now.

**Review by bkrem — APPROVED** [review]

**Comment by bkrem** [comment]
> I'm gonna talk to some polkadot wallet teams to implement the wallet-connect on their side. If any changes happen in the RPCs I will send PRs to keep these up-to-date

Legend 💪

**Comment by bkrem** [comment]
Good to go from my side, all good for you @IljaDaderko?

**Comment by IljaDaderko** [comment]
@bkrem Yep good to go as well, only concern is failing preview builds for various dapp flavours, but that could be unrelated to this pr?

**Comment by bkrem** [comment]
> only concern is failing preview builds for various dapp flavours, but that could be unrelated to this pr?

It's just Vercel gating deployments, so I had to authorize the two relevant ones here (`react-dapp-v2`, `react-wallet-v2`). The other dapp examples need a proper overhaul anyways once we have providers updated for v2 👍

**Pull request merged**
